build_runner/lib/src/generate/build_definition.dart

```diff
@@ -239,4 +246,63 @@ class _Loader {
       yield new AssetId(package, path);
     }
   }
+
+  /// Handles cleanup of pre-existing outputs for initial builds (where there is
+  /// no cached graph).
+  Future<Null> _initialBuildCleanup(Set<AssetId> conflictingAssets,
```

Comment:
one of my goals with `BuildDefinition` was to have a more clear separation between figuring out what work needs to happen and doing the work.

I'm somewhat torn on this change.

Reply:
We are already doing a fair bit of work, because we are tracking the asset graph here. Because of that we have to check the build script for updates here (because we might have to throw away the asset graph). In order to do that properly we have to update the asset graph as well so it contains any new sources.

In terms of the `_initialBuildCleanup` I think that still makes sense here as well as its a part of setting up the initial state for the build. The state of the world is not valid until those files have been cleaned up. This also allows us to garbage collect the `updates` and `conflictingOutputs` sets which previously never could be.

We are still not doing any actual building as a part of this change, we are only setting up the state of the world such that it is ready for a build to happen.

Reply:
Should we look at a new name for `load`? Maybe `prepareWorkspace` or something?

Reply:
Sure, I would be fine w/that